### DISCUSSION TITLE
Problem with finding 'client_secret.json' file

### DISCUSSION THREAD

**Discussion by airfinland** [gilesknap/gphotos-sync#456] [Q&A]
Hi all,

I'm on MacOS 13.6.3 running Python 3.11 as I couldn't install 'gphotos-sync' on the latest version of Python so downgraded. Followed the instructions on 'Creating an OAuth Client ID' and extracted 'client_secret.json' file and after that I successfully installed 'gphotos-sync' on virtual environment.

1. ghphotos-sync --version is working and command returns version result of 3.1.3.

When I'm trying to run 'gphotos-sync' it returns with error _FileNotFoundError: [Errno 2] No such file or directory: '/Users/x/Library/Application Support/gphotos-sync/client_secret.json'_ which is weird as the 'client_secret.json' file is placed in the proper folder as stated in the initial setup process i.e., _~/Library/Application Support/gphotos-sync/client_secret.json_.

It really doesn't matter is the folder I want to extract to a portable device or my computer internal hard drive as the error is the same.

Below is the log file the program creates in the folder I want to extract the photos to:

> 12-17 10:23:28 gphotos_sync.Main WARNING  gphotos-sync 3.1.3 2023-12-17 10:23:28.679493
> 12-17 10:23:28 gphotos_sync.Utils DEBUG    MINIMUM_DATE = 1800-01-01 00:00:00
> 12-17 10:23:28 gphotos_sync.Checks INFO     Target filesystem /volumes/portable/google is hfs
> 12-17 10:23:28 gphotos_sync.Checks DEBUG    Checking if is filesystem supports symbolic links...
> 12-17 10:23:28 gphotos_sync.Checks DEBUG    attempting to symlink /Volumes/Portable/Google/test_src_3373367887 to /Volumes/Portable/Google/test_dst_145614603
> 12-17 10:23:28 gphotos_sync.Checks DEBUG    Checking if File system supports unicode filenames...
> 12-17 10:23:28 gphotos_sync.Checks INFO     Filesystem supports Unicode filenames
> 12-17 10:23:28 gphotos_sync.Checks DEBUG    Checking if File system is case insensitive...
> 12-17 10:23:28 gphotos_sync.Checks INFO     Case insensitive file system found
> 12-17 10:23:28 gphotos_sync.Checks INFO     Max Path Length: 1024
> 12-17 10:23:28 gphotos_sync.Checks INFO     Max filename length: 255
> 12-17 10:23:28 gphotos_sync.Main INFO     version: 3.1.3, database schema version 5.7
> 12-17 10:23:28 gphotos_sync.LocalData INFO     Saving Database ...
> 12-17 10:23:28 gphotos_sync.LocalData INFO     Database Saved.
> 12-17 10:23:28 gphotos_sync.LocalData INFO     Saving Database ...
> 12-17 10:23:28 gphotos_sync.LocalData INFO     Database Saved.
> 12-17 10:23:28 gphotos_sync.Main ERROR    
> Process failed.
> Traceback (most recent call last):
>   File "/Users/x/venv/lib/python3.11/site-packages/gphotos_sync/authorize.py", line 50, in __init__
>     with secrets_file.open("r") as stream:
>          ^^^^^^^^^^^^^^^^^^^^^^
>   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/pathlib.py", line 1044, in open
>     return io.open(self, mode, buffering, encoding, errors, newline)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> FileNotFoundError: [Errno 2] No such file or directory: '/Users/x/Library/Application Support/gphotos-sync/client_secret.json'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/x/venv/lib/python3.11/site-packages/gphotos_sync/Main.py", line 506, in main
>     self.setup(args, db_path)
>   File "/Users/x/venv/lib/python3.11/site-packages/gphotos_sync/Main.py", line 346, in setup
>     self.auth = Authorize(
>                 ^^^^^^^^^^
>   File "/Users/x/venv/lib/python3.11/site-packages/gphotos_sync/authorize.py", line 64, in __init__
>     exit(1)
>   File "<frozen _sitebuiltins>", line 26, in __call__
> SystemExit: 1
> 12-17 10:23:28 gphotos_sync.Main WARNING  Done.

Hope somebody can help me solve this issue - I've tried to find solutions but no luck yet.

**Comment by gilesknap** [comment]
I assume from this your username is `x`.

Is the JSON file at `/Users/x/Library/Application Support/gphotos-sync/client_secret.json`
If not copy it there and all should be well.

In many shells the `~` should resolve to the root of your user folder - in this case  `/Users/x/` but depending on the shell you are using that may not have worked.

Thanks for reporting that gphotos-sync does not work on python 3.12 - time for a new release.

**Comment by airfinland** [comment]
The thing is that the 'client_secret.json' file is in the correct directory (the one you stated and found in initial setup tutorial) but it seems that the program does not recognise it.

I followed the instructions for OAuth when I created the .json file and it should be valid by all means and not corrupted/incorrect.

**Comment by gilesknap** [comment]
It's not saying the file is corrupt, just that it cannot find it.

try cutting and pasting this command into a shell to make sure there is not a subtle difference between the filename you can see and the one the code is looking for:

```bash
ls '/Users/x/Library/Application Support/gphotos-sync/client_secret.json'
```

**Comment by airfinland** [comment]
Executing the command you suggested returns the one below:

> No such file or directory

However, executing `ls '/Library/Application Support/gphotos-sync/client_secret.json'` returns 'client_secret.json'.

Could this be some sort of root issue?

**Comment by airfinland** [comment]
I solved the issue - looks like I had the '/Users/x/Library' hidden for some reason and placed the 'client_secret.json' in the wrong folder. All good now and program is working. Apologies for this.

**Comment by gilesknap** [comment]
Thanks for letting me know. Closing.
